Tests/Crypto/SPIDiffieHellmanTest.m

```diff
@@ -32,7 +32,7 @@ - (void)testExchange {
     JKBigInteger *primeG = [[JKBigInteger alloc] initWithString:@"2"];
     
     JKBigInteger *privateKeyA = [SPIDiffieHellman randomPrivateKey:primeP];
-    JKBigInteger *privateKeyB = [SPIDiffieHellman randomPrivateKey:primeG];
+    JKBigInteger *privateKeyB = [SPIDiffieHellman randomPrivateKey:primeP];
```

Comment:
I believe this was wrong, as primeG is only 2. This caused private & public keys to be <2 and the test to fail now that we have the check.

Reply:
Both g and p are static. g (i.e. the generator) should always be 2. And p is the "prime modulus" which is always set to the same special large number. If it helps, further detail on this is on slide 6 of https://docs.google.com/presentation/d/1-IyRnjJoyjLTOcpWZS4928yHWbf6hX11wUZ0pUFFJh8/edit#slide=id.g22e8e426d90_0_39.

Reply:
Yep I think thats all being done currently, just the generation of a random private key was resulting in 1 for privateKeyB. The parameter we pass in to the function randomPrivateKey() is a max integer value, and the functions generates a random value between 0 and that value. So I think we were just passing in the wrong value to this function